TwirfonyBundle/Controller/TwirpController.php

```diff
@@ -3,23 +3,31 @@
 namespace Twirfony\TwirfonyBundle\Controller;
 
 use Throwable;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Google\Protobuf\Internal\Message;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Twirfony\TwirfonyBundle\Event\ErrorEvent;
 use Twirfony\TwirfonyBundle\Event\RequestEvent;
 use Twirfony\TwirfonyBundle\Event\ResultEvent;
 use Twirfony\TwirpError;
 
-class TwirpController extends Controller
+class TwirpController extends AbstractController implements LoggerAwareInterface
 {
+    private $logger;
+
+    public function setLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
 
     public function rpcAction(Request $request)
     {
         $inputType = $request->attributes->get('inputType');
         $serviceId = $request->attributes->get('service');
-        $service = $this->get($serviceId);
+        $service = $this->container->get($serviceId);
```

Comment:
I'm not familiar with this functionality, but I'm curious, what services is this likely to be getting? 

I thought the container within an AbstractController was only able to access a certain subset of services.

Reply:
This fetches the resolver service assigned to the rpc endpoint being served. This resolver service implements the endpoint method itself. These resolver services are defined by the consumers of this lib, and are somewhat dynamic (they are tagged with `twirp.service`).

The default container made available to AbstractController is indeed a limited one, however we now override it with the full container when setting up TwirpController via `- ["setContainer", ["@service_container"]]`.

There is probably a way to be smarter about injecting the resolver services dynamically so we don't need the full container, but I haven't thought about it too much.